File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int temp = storey;
        int cnt = 0;

        while (temp > 0) {
            int remain = temp % 10;  // 현재 처리할 나머지
            temp /= 10;  // 다음 자릿수로 이동

            if (remain > 5) {
                // 반올림 처리
                temp++;
                cnt += 10 - remain;
            } else if (remain == 5) {
                // 5는 다음 자릿수를 보고 반올림 또는 내림 처리
                if (temp % 10 >= 5) {
                    temp++;
                    cnt += 10 - remain;
                } else {
                    cnt += remain;
                }
            } else {
                // 5 미만인 경우 그대로 내림 처리
                cnt += remain;
            }
        }
        return cnt;
    }
}

File: 프로그래머스/2/148653. 마법의 엘리베이터/마법의 엘리베이터.kt
class Solution {
    fun solution(storey: Int): Int {
        var temp = storey
        var cnt = 0

        while (temp > 0) {
            //제일 작은 자릿 수부터 계산하도록 10으로 나눈다.
            var remain = temp % 10
            //temp 값은 작은 자릿 수를 빼야하니 10으로 나눈 몫으로 변경한다.
            temp /= 10

            when {
                remain > 5 -> { //반올림 처리
                    temp ++
                    cnt += 10 - remain
                }
                remain == 5 -> { //끝자리가 5일 수도 있고, 0일 수도 있다.
                    if (temp % 10 >= 5) {
                        temp ++
                        cnt += 10 - remain
                    } else {
                        cnt += remain
                    }
                }
                else -> {
                    cnt += remain
                }
            }
        }
        return cnt
    }
}